dace/codegen/tools/type_inference.py

```diff
@@ -405,6 +405,9 @@ def _infer_dtype(t: Union[ast.Name, ast.Attribute]):
 
 def _Attribute(t, symbols, inferred_symbols):
     inferred_type = _dispatch(t.value, symbols, inferred_symbols)
+    if (isinstance(inferred_type, dtypes.pointer) and isinstance(inferred_type.base_type, dtypes.struct) and
```

Comment:
@alexnick83 what about plain structs (i.e., not pointers)

Reply:
Since we have decided that Structures will always resolve to a pointer to a struct, I think it is better to not handle that case (now), since it might be something unrelated to nested data and cause further confusion.

Reply:
Note: This assumption was valid for the time being, but I am now arguing to change that assumption. For cases where the type of an expression is being evaluated, where the `t.value` type of an Attribute expression is not merely a structure, but an array of structures (indexed through a Slice), the returned `inferred_type` is no longer a pointer but directly a plain struct.

Example: `my_array_of_structs[i].my_attribute`

Commit https://github.com/spcl/dace/commit/e2561ed5bda44808b910940093728de808213508 implements a handle for this additional case.